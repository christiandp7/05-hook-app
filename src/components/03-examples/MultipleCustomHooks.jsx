import React from 'react'
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';
import '../02-useEffect/effect.css';

const MultipleCustomHooks = () => {

  const  { counter, increment } = useCounter(1)

  const {loading, data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`)
  // console.log(state);
  const { author, quote } = !!data && data[0]

  return (
    <div>
      <h1>Breaking Bad Quotes</h1>
      <hr/>
      {
        loading ?
        (<div className="aler alert-info text-center">
          loading
        </div>)
        :
        (<blockquote className="blockquote text-right">
          <p className="text-italic">{quote}</p>
          <footer className="blockquote-footer">{author}</footer>
        </blockquote>)
      }

      <button 
        onClick={ () => increment(1) }
        className="btn btn-primary">Siguiente Quote</button>
      
      
    </div>
  )
}

export default MultipleCustomHooks