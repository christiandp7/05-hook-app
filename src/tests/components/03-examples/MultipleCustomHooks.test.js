import React from 'react'
import { shallow } from 'enzyme'
import MultipleCustomHooks from '../../../components/03-examples/MultipleCustomHooks'
import { useFetch } from '../../../hooks/useFetch'
import { useCounter } from '../../../hooks/useCounter'

jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')

describe('Tests in <MultipleCustomHooks />', () => {
	test('should render', () => {
		useCounter.mockReturnValue({
			counter: 1,
			increment: () => {},
		})
		useFetch.mockReturnValue({
			data: null,
			loading: true,
			error: null,
		})

		const wrapper = shallow(<MultipleCustomHooks />)
		expect(wrapper).toMatchSnapshot()
	})

	test('should show the info', () => {
		useCounter.mockReturnValue({
			counter: 1,
			increment: () => {},
		})
		useFetch.mockReturnValue({
			data: [
				{
					author: 'Christian',
					quote: 'Hola Mundo',
				},
			],
			loading: false,
			error: null,
		})
		const wrapper = shallow(<MultipleCustomHooks />)
		expect(wrapper.find('.alert').exists()).toBe(false)
		expect(wrapper.find('.text-italic').text().trim()).toBe('Hola Mundo')
		expect(wrapper.find('footer').text().trim()).toBe('Christian')
		// console.log(wrapper.html())
	})
})
