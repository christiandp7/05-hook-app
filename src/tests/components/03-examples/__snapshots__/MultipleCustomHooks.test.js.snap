// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <MultipleCustomHooks /> should render 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="aler alert-info text-center"
  >
    loading
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Quote
  </button>
</div>
`;
